homework-submission/App.js

```diff
@@ -0,0 +1,105 @@
+const path = '/data.json';
+
+const ListItem = ({ id, description, deadline, className }) => {
+  return (
+    <li className={className}>
+      Todo List Number: {id} Description: {description} Deadline: {deadline}
+    </li>
+  );
+};
+
+class TodoApp extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: [],
+    };
+  }
+
+  componentDidMount() {
+    readFile(path).then(response => {
+      this.setState(() => ({
+        todos: response,
+      }));
+    });
+  }
+
+  render() {
+    const todos = this.state.todos;
+    const NewTodo = {
+      id: 0,
+      description: '',
+      deadline: '',
+      done: false,
+    };
+    let TodoNumber = 0;
+
+    return (
+      <div>
+        <h2>TodoLists</h2>
+        <ul>
+          {todos.map(({ id, description, deadline, done }, i) => (
+            <ListItem
+              id={id}
+              description={description}
+              deadline={deadline}
+              className={done === true ? 'strike_through' : ''}
+              key={i}
+            />
+          ))}
+        </ul>
+        <form>
+          <input
+            type="text"
+            name="todo_description"
+            placeholder="Give a description"
+            onChange={event => {
+              NewTodo.description = event.target.value;
+            }}
+          />
+          <input
+            type="text"
+            name="todo_deadline"
+            placeholder="Give a deadline"
+            onChange={event => {
+              NewTodo.deadline = event.target.value;
+            }}
+          />
+          <input
+            type="submit"
+            value="Add"
+            onClick={event => {
```

Comment:
it would be very complicated in larger apps adding more in line event functions,
perhaps isolating them in a new method could be more legible, shorter and easier to follow.

Reply:
Functions are listed seperately and reference is made while using them.